### PR TITLE
Fix building on OpenBSD 7.4

### PR DESCRIPTION
**Pull request by hgojf**
I tried to build this on OpenBSD 7.4 but ran into a few errors, these patches allow it to build.

### CODE REVIEW

File: wscript [hunk 1]
```diff
@@ -405,6 +408,9 @@ def configure(conf):
 		else:
 			for i in a:
 				conf.check_cc(lib = i)
+	elif conf.env.DEST_OS == 'openbsd':
+		#libdl is a part of libc on this platform
```
Comment (nekonomicon) [inline wscript:412]: linkage works with `conf.check_cc(lib='dl', mandatory=False)`?

Comment (hgojf) [reply wscript:412]: Yes that works

Comment (a1batross) [reply wscript:412]: Could you revert this and add `mandatory=False` flag to libdl check then? I don't think making libdl optional will hurt anything.

**Comment by a1batross** [comment]
Other than `libdl` check, I'm fine with these changes.

**Comment by hgojf** [comment]
Applied that change

**Review by a1batross — APPROVED** [review]

**Comment by a1batross** [comment]
Thank you! <3

**Pull request merged**